addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -192,7 +192,15 @@ def _l10n_es_edi_facturae_inv_lines_to_items(self, conversion_rate=None):
                 continue
             invoice_line_values = {}
 
-            price_before_discount = line.currency_id.round(line.price_subtotal / (1 - line.discount / 100.0))
+            tax_base_before_discount = self.env['account.tax']._convert_to_tax_base_line_dict(
+                base_line=line,
+                currency=line.currency_id,
+                taxes=line.tax_ids,
+                price_unit=line.price_unit,
+                quantity=line.quantity,
+            )
+            tax_before_discount = self.env['account.tax']._compute_taxes([tax_base_before_discount])
+            price_before_discount = sum(to_update['price_subtotal'] for _dummy, to_update in tax_before_discount['base_lines_to_update'])
             discount = max(0., (price_before_discount - line.price_subtotal))
             surcharge = abs(min(0., (price_before_discount - line.price_subtotal)))
             totals['total_gross_amount'] += line.price_subtotal
```

Comment:
```suggestion
            totals['total_gross_amount'] += price_before_discount
```
I just realised during testing that that line was wrong too (testing with 2 lines at 100 + 21%taxes and one of these with a 100% discount)
as it should would generate this
```xml
<TotalGrossAmount>200.00</TotalGrossAmount>  <!-- Before that change it would be 100, which is wrong -->
<TotalGeneralDiscounts>100.00</TotalGeneralDiscounts>
<TotalGrossAmountBeforeTaxes>100.00</TotalGrossAmountBeforeTaxes>  <!-- Before that change it would be 0, which is wrong -->
<TotalTaxOutputs>21.00</TotalTaxOutputs>
<TotalTaxesWithheld>0.00</TotalTaxesWithheld>
<InvoiceTotal>121.00</InvoiceTotal>
<TotalOutstandingAmount>121.00</TotalOutstandingAmount>
<TotalExecutableAmount>121.00</TotalExecutableAmount>
```

Other than that it works perfectly :)

Reply:
@JulienAlardot Changes are done!!